src/main/java/no/digipost/monitoring/keystore/CertificateExpiryMetrics.java

```diff
@@ -0,0 +1,129 @@
+import java.security.cert.X509Certificate;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.Objects;
+
+import static java.time.temporal.ChronoUnit.DAYS;
+
+/**
+ * Micrometer {@link MeterBinder} registering gauges for Java key store certificates.
+ *
+ * <p>
+ * Each certificate gauge keeps track of seconds until certificate expiry. It
+ * also has some useful tags such as keystore alias, X509 certificate fields as
+ * issuer, subject, notBefore and notAfter.
+ */
+public final class CertificateExpiryMetrics implements MeterBinder {
+
+    public static final int DEFAULT_DAYS_TO_EXPIRY_WARN_THRESHOLD = 60;
+
+    private final KeyStore keyStore;
+    private Clock clock;
+    private final int days_to_expiry_warn_threshold;
+
+    /**
+     * Create a new meter binder.
+     *
+     * @param keyStore key store containing certificates to monitor.
+     */
+    public CertificateExpiryMetrics(KeyStore keyStore, Clock clock) {
+        this(keyStore, clock, DEFAULT_DAYS_TO_EXPIRY_WARN_THRESHOLD);
+    }
+
+    /**
+     * Create a new meter binder setting days to warn
+     *
+     * @param keyStore key store containing certificates to monitor.
+     * @param clock The time of the server
+     */
+    public CertificateExpiryMetrics(KeyStore keyStore, Clock clock, int days_to_expiry_warn_threshold) {
+        this.clock                         = clock;
+        this.days_to_expiry_warn_threshold = days_to_expiry_warn_threshold;
+        Objects.requireNonNull(keyStore);
+        this.keyStore = keyStore;
+    }
+
+    @Override
+    public void bindTo(MeterRegistry registry) {
+        try {
+            for (Enumeration<String> aliases = keyStore.aliases(); aliases.hasMoreElements(); ) {
+                String alias = aliases.nextElement();
+                registerGauge(registry, alias);
+            }
+        } catch (KeyStoreException e) {
+            throw new IllegalStateException(e);
+        }
+
+    }
+
+    private void registerGauge(MeterRegistry registry, String alias) throws KeyStoreException {
+        X509Certificate cert = (X509Certificate) keyStore.getCertificate(alias);
+        Tag[] tags = new Tag[]{
+                Tag.of("alias", alias),
+                Tag.of("issuer", cert.getIssuerX500Principal().getName()),
+                Tag.of("subject", cert.getSubjectX500Principal().getName()),
+                Tag.of("notAfter", cert.getNotAfter().toString()),
+                Tag.of("notBefore", cert.getNotBefore().toString())
+        };
+        registry.gauge(
+                "cert_expiry", Arrays.asList(tags), cert,
+                c -> Duration.between(Instant.now(clock), c.getNotAfter().toInstant()).getSeconds()
+        );
+        registry.gauge("certificates_status", Arrays.asList(tags), cert,
```

Comment:
Like greit å bare sette opp disse som en liste from the get-go? A.k.a noe sånt:
```suggestion
        List<Tag> tags = Arrays.asList(
                Tag.of("alias", alias),
                Tag.of("issuer", cert.getIssuerX500Principal().getName()),
                Tag.of("subject", cert.getSubjectX500Principal().getName()),
                Tag.of("notAfter", cert.getNotAfter().toString()),
                Tag.of("notBefore", cert.getNotBefore().toString())
        );
        registry.gauge(
                "cert_expiry", tags, cert,
                c -> Duration.between(Instant.now(clock), c.getNotAfter().toInstant()).getSeconds()
        );
        registry.gauge("certificates_status", tags, cert,
```

Reply:
jaja. Seff. Forcepushet.